sql4.sql

```diff
@@ -0,0 +1,178 @@
+LEFT JOIN ORDERDETAILS OD ON O.ORDERID=OD.ORDERID 
+LEFT JOIN CUSTOMER C ON C.CUSTOMERID=O.CUSTOMERID
+LEFT JOIN PROD P ON P.PRODUCTID=OD.PRODUCTID;
+
+-- 6. Show all products that have never been ordered, along with their category information.
+SELECT P.PRODUCTNAME,C.CATEGORYNAME FROM PROD P
+LEFT JOIN CATEGORIES C ON P.CategoryID=C.CategoryID
+WHERE P.PRODUCTID NOT IN (SELECT PRODUCTID FROM ORDERDETAILS);
+
+-- 7. List all customers who have placed orders in the last week, along with the products they've purchased.
+
+SELECT C.CustomerID,C.CustomerName,P.ProductName
+FROM Customer C
+LEFT JOIN Ord O ON C.CustomerID = O.CustomerID
+LEFT JOIN OrderDetails OD ON O.OrderID = OD.OrderID
+LEFT JOIN Prod P ON P.ProductID = OD.ProductID
+WHERE DATEPART(WEEK, O.OrderDate) = DATEPART(WEEK, GETDATE()) - 1;
+    
+
+-- 8. Display all categories with products priced over $100, including categories without such products.
+SELECT C.CATEGORYNAME,P.PRODUCTNAME ,P.PRICE
+FROM CATEGORIES C
+LEFT JOIN PROD P ON P.CATEGORYID=C.CATEGORYID
+WHERE P.PRODUCTNAME IS NULL OR P.PRICE>100
+
+-- 9. Show all orders placed before 2023 and any associated product information.
+SELECT O.ORDERID,O.ORDERDATE FROM ORD O
+JOIN ORDERDETAILS OD ON OD.ORDERID=O.ORDERID
+JOIN PROD P ON P.PRODUCTID=OD.PRODUCTID
+WHERE YEAR(O.ORDERDATE) < 2023;
+
+-- 10. List all possible category-customer combinations, regardless of whether the customer has purchased a product from that category.
+
+SELECT C.CATEGORYNAME,P.CUSTOMERNAME 
+FROM CATEGORIES C
+CROSS JOIN CUSTOMER P
+
+-------------------------------------------------------------------------------------------------------------------------------
+
+-- 1. List all books along with their authors, including books without assigned authors.
+SELECT B.TITLE,A.AUTHORNAME
+FROM BOOKS B
+LEFT JOIN AUTHORS A ON B.AUTHORID=A.AUTHORID;
+
+-- 2. Display all patrons and their loan history, including patrons who have never borrowed a book.
+SELECT P.PATRONNAME,L.LOANDATE,L.RETURNDATE
+FROM PATRONS P
+LEFT JOIN LOANS L ON P.PATRONID=L.PATRONID;
+
+-- 3. Show all authors and the books they've written, including authors who haven't written any books in our collection.
+SELECT A.AUTHORNAME, B.TITLE 
+FROM AUTHORS A 
+LEFT JOIN BOOKS B ON A.AUTHORID=B.AUTHORID;
+
+-- 4. List all possible book-patron combinations, regardless of whether a loan has occurred.
+SELECT P.PATRONNAME, B.TITLE
+FROM PATRONS P
+CROSS JOIN BOOKS B;
+
+-- 5. Display all loans with book and patron information, including loans where either the book or patron information is missing.
+SELECT L.LOANDATE,L.RETURNDATE, B.TITLE,B.PUBLICATIONYEAR,P.PATRONNAME
+FROM LOANS L
+LEFT JOIN BOOKS B ON B.BOOKID=L.BOOKID
+LEFT JOIN PATRONS P ON P.PATRONID=L.PATRONID;
+
+-- 6. Show all books that have never been loaned, along with their author information.
+SELECT B.TITLE,A.AUTHORNAME
+FROM BOOKS B
+LEFT JOIN LOANS L ON B.BOOKID= L.BOOKID
+JOIN AUTHORS A ON A.AUTHORID=B.AUTHORID
```

Comment:
Ensure JOIN with authors is correctly referencing authors without books.

Reply:
updated with left join